src/errors.rs

```diff
@@ -0,0 +1,63 @@
+use derive_more::Display;
+#[cfg(feature = "std")]
+use derive_more::Error;
+
+macro_rules! impl_error {
```

Comment:
I think `Error -> &str` is semantically incorrect. Previously it was common to use `Error::description`, but now it's deprecated in favor of `Display`. Thus, instead, I prefer to have a separate `fn as_str()`.

Reply:
fixed